addons/controllers/antrea/antreaconfig_util.go

```diff
@@ -295,6 +325,69 @@ func mapAntreaConfigSpec(cluster *clusterv1beta1.Cluster, config *cniv1alpha2.An
 	if semver.Compare(version, "v1.9.0") >= 0 {
 		configSpec.Antrea.AntreaConfigDataValue.FeatureGates.TopologyAwareHints = &config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.TopologyAwareHints
 	}
+	// NSX related
+	if semver.Compare(version, "1.9.0") >= 0 && config.Spec.AntreaNsx.Enable {
+		configSpec.AntreaNsx.Enable = config.Spec.AntreaNsx.Enable
+		if config.Spec.AntreaNsx.AntreaNsxConfig.BootstrapFrom == bootstrapFromInline {
+			configSpec.AntreaNsx.AntreaNsxConfig.NSXManagers = config.Spec.AntreaNsx.AntreaNsxConfig.NSXManagers
+			configSpec.AntreaNsx.AntreaNsxConfig.ClusterName = config.Spec.AntreaNsx.AntreaNsxConfig.ClusterName
+			// NSX cert
+			secret, err := getNSXCert(client, config.Name, config.Namespace)
+			if err != nil {
+				return configSpec, err
+			}
+			configSpec.AntreaNsx.AntreaNsxConfig.NSXCert = string(secret.Data["tls.crt"])
+			configSpec.AntreaNsx.AntreaNsxConfig.NSXKey = string(secret.Data["tls.key"])
+			configSpec.AntreaNsx.AntreaNsxConfig.VPCPath = config.Spec.AntreaNsx.AntreaNsxConfig.VPCPath
+		} else {
+			configSpec.AntreaNsx.AntreaNsxConfig.BootstrapFrom = bootstrapFromSupervisorCluster
+			configSpec.AntreaNsx.AntreaNsxConfig.BootstrapSupervisorResourceName = getNSXServiceAccountName(cluster.Name)
+		}
+		configSpec.AntreaNsx.AntreaNsxConfig.ProxyEndpoints.NSXRpcFwdProxy = config.Spec.AntreaNsx.AntreaNsxConfig.ProxyEndpoints.NSXRpcFwdProxy
+		configSpec.AntreaNsx.AntreaNsxConfig.ProxyEndpoints.RestApi = config.Spec.AntreaNsx.AntreaNsxConfig.ProxyEndpoints.RestApi
+
+		ccpConf := config.Spec.AntreaNsx.AntreaNsxConfig.CcpAdapterConf
+		if err := copyStructAtoB(ccpConf, &configSpec.AntreaNsx.AntreaNsxConfig.CcpAdapterConf); err != nil {
+			return configSpec, err
+		}
+		mpConf := config.Spec.AntreaNsx.AntreaNsxConfig.MpAdapterConf
+		if err := copyStructAtoB(mpConf, &configSpec.AntreaNsx.AntreaNsxConfig.MpAdapterConf); err != nil {
+			return configSpec, err
+		}
+	}
 
 	return configSpec, nil
 }
+
+func copyStructAtoB(a interface{}, b interface{}) error {
+	va := reflect.ValueOf(a)
```

Comment:
reflect can make it more feasible, but at same time it can introduce performance cost, maybe consider if the implementation is OK.